src/main/java/com/minecolonies/coremod/network/messages/ChangeFreeToInteractBlockMessage.java

```diff
@@ -0,0 +1,153 @@
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Message to execute the renaiming of the townHall.
+ */
+public class ChangeFreeToInteractBlockMessage extends AbstractMessage<ChangeFreeToInteractBlockMessage, IMessage>
+{
+
+    /**
+     * Enums for Message Type for the freeBlock message.
+     * <p>
+     * ADD_BLOCK       Add a block or pos.
+     * REMOVE_BLOCK    Removing a block or pos.
+     */
+    public enum MessageType
+    {
+        REMOVE_BLOCK,
+        ADD_BLOCK,
+    }
+
+    /**
+     * The id of the colony.
+     */
+    private int colonyId;
+
+    /**
+     * The position of the free to interact block.
+     */
+    private BlockPos pos = new BlockPos(0, 0 ,0);
+
+    /**
+     * The blockState which can be freely interacted with.
+     */
+    private Block block = Blocks.DIRT;
+
+    /**
+     * The type of the message.
+     */
+    private MessageType type;
+
+    /**
+     * Empty public constructor.
+     */
+    public ChangeFreeToInteractBlockMessage()
+    {
+        super();
+    }
+
+    /**
+     * Message creation to add a new freely interactable block to the colony.
+     *
+     * @param colony Colony the block can be interacted with in.
+     * @param block the blockState.
+     * @param type the type of message.
+     */
+    public ChangeFreeToInteractBlockMessage(@NotNull final ColonyView colony, @Nullable final Block block, @NotNull final MessageType type)
+    {
+        super();
+        this.colonyId = colony.getID();
+        this.pos = new BlockPos(0,0,0);
+        this.block = block;
+        this.type = type;
+    }
+
+    /**
+     * Message creation to add a new freely interactable position to the colony.
+     *
+     * @param colony Colony the position can be interacted with in.
+     * @param pos the position.
+     * @param type the type of message.
+     */
+    public ChangeFreeToInteractBlockMessage(@NotNull final ColonyView colony, @Nullable final BlockPos pos, @NotNull final MessageType type)
+    {
+        super();
+        this.colonyId = colony.getID();
+        this.pos = pos;
+        this.block = Blocks.DIRT;
+        this.type = type;
+    }
+
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        colonyId = buf.readInt();
+        block = Block.getBlockFromName(ByteBufUtils.readUTF8String(buf));
+        pos = BlockPosUtil.readFromByteBuf(buf);
+        type = MessageType.values()[buf.readInt()];
+    }
+
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        buf.writeInt(colonyId);
+        ByteBufUtils.writeUTF8String(buf, block.getRegistryName().toString());
+        BlockPosUtil.writeToByteBuf(buf, pos);
+        buf.writeInt(type.ordinal());
+    }
+
+    @Override
+    public void messageOnServerThread(final ChangeFreeToInteractBlockMessage message, final EntityPlayerMP player)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "messageOnServerThread" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)